File: Queue/Queue Linked List.cpp
#include <iostream>
using namespace std;
struct Node{
	int data;
	Node *next;
	Node(int d){
		data = d;
		next =NULL;
	}
};
class Queue{
	Node *rear;
	Node *front;
	int value(){
		int v;
		cout<< "Enter the Value : ";cin>>v;
		return v;
	}
	public :
		Queue(){
			front = NULL;
			rear = NULL;
		}
		
		void enqueue(){
			if(front==NULL){
				front = new Node(value());
				rear =  front;
			}
			else{
				Node *temp = new Node(value());
				rear->next = temp;
				rear = temp;
			}
		}
		void dequeue(){
			if(front==NULL){
				cout<< "Under Flow Queue"<<endl;
			}
			else{
				cout << "Dequeue Value is : "<<front->data<<endl;
				front = front->next;
				if(front == NULL){
					rear = NULL;
				}
			}
		}
		void peek(){
			if (front==NULL){
				cout << "Empty Queue "<<endl;
			}
			else{
				cout<< "Peek Value is : "<<front->data<<endl;
			}
		}
		void display(){
			if(front==NULL){
				cout <<"Empty Queue"<<endl;
				return ;
			}
			cout << "Queue Linked List"<<endl;
			Node *temp = front;
			while(temp!=NULL){
				cout << temp->data<<"   ";
				temp = temp->next;
			}
			cout<<endl;
		}
};
int main(){
	Queue q;
	int c,a;
while(1){
        cout << "1.EXIT"<<endl;
		cout << "2.ENQUEUE"<<endl;
		cout << "3.DEQUEUE"<<endl;
		cout << "4.PEEK"<<endl; 	
		cout << "5.Display"<<endl;
        cout << "Enter your Choice : ";cin>>c;
switch (c)
	{
		case 1:{
			return 0;
			break;
		}
		case 2:{
			q.enqueue();
			break;
		}
		case 3:{
			q.dequeue();
			break;
		}
		case 4:{
		    q.peek();
			break;
		}
		case 5:{
			q.display();
			break;
		}
		default :{
			cout << "Wrong Input! "<<endl;
			break;
		}
	}
  }
	return 0;
}

File: DataStructures/src/QueueLinkedList/QueueLinkedList.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package QueueLinkedList;

import java.util.Scanner;

/**
 *
 * @author Naeem ur Rahman
 */
public class QueueLinkedList {

    private Scanner i = new Scanner(System.in);

    private class node {

        int data;
        node next;

        node(int d) {
            this.data = d;
            this.next = null;
        }
    }
    private node rear;
    private node front;

    private int value() {
        int v = 0;
        System.out.print("Enter the Value : ");
        v = i.nextInt();
        return v;
    }

    public void display() {
        if (front == null) {
            System.out.println("Empty Queue !");
        } else {
            node ptr = front;
            while (ptr != null) {
                System.out.print(ptr.data + "   ");
                ptr = ptr.next;
            }
            System.out.println("");
        }
    }

    public void enqueue() {
        if (front == null) {
            front = new node(value());
            rear = front;
        } else {
            node ptr = new node(value());
            rear.next = ptr;
            rear = ptr;
        }
    }

    public void dequeue() {
        if (front == null) {
            System.out.println("UnderFlow Queue LinkedList ! ");
        } else {
            System.out.println("Deleted Data is : "+ front.data);
            front = front.next;
        }
    }
    public int count(){
        int c =0;
        if(front==null){
            return c;
        }
        node ptr = front;
        while(ptr!=null){
            c++;
            ptr=ptr.next;
        }
        return c;
    }
}
